src/core/services/ripple/ripple.js

```diff
@@ -307,17 +345,71 @@ InkRippleCtrl.prototype.clearRipples = function () {
     this.fadeInComplete(this.ripples[ i ]);
   }
 };
-
+/**
+ * Check the md-ink-explode-dur attribute for modification and update the duration
+ * @returns {*}
+ */
+InkRippleCtrl.prototype.updateDuration = function() {
+  this.duration = parseInt(this.$$element.attr('md-ink-explode-dur')) || DURATION;
+  this.fadeInDuration = (this.explode) ? this.duration : this.duration*.35;
+  return this.duration;
+};
 /**
  * Creates the ripple container element
  * @returns {*}
  */
 InkRippleCtrl.prototype.createContainer = function () {
-  var container = angular.element('<div class="md-ripple-container"></div>');
-  this.$element.append(container);
+  var container = this.getContainer();
```

Comment:
The reasoning behind this is that one element with many different `md-ink-explode` references to it can have many `<div class="md-ripple-container"></div>` elements. Which I guess is fine, just seems unnecessary. The other reason is to observe changes in the container attribute. 
So here (with the old createContainer function) there are three containers, which correspond to each of the three list items. 
![multiple containers](https://cloud.githubusercontent.com/assets/7891759/14412703/0f7b3d82-ff2e-11e5-9677-cd3f6be78653.PNG)

This works fine either way, it might be less expensive to keep the old function then to look for an existing container.